lib/platform/github/index.ts

```diff
@@ -1719,3 +1731,117 @@ export async function getVulnerabilityAlerts(): Promise<VulnerabilityAlert[]> {
+  message: string
+): Promise<string> {
+  await pushCommitToRenovateRef(localCommitSha, branchName);
+  const treeItems = await listCommitTree(localCommitSha);
+
+  const treeRes = await githubApi.postJson<{ sha: string }>(
+    `/repos/${config.repository}/git/trees`,
+    { body: { tree: treeItems } }
+  );
+  const treeSha = treeRes.body.sha;
+
+  const commitRes = await githubApi.postJson<{ sha: string }>(
+    `/repos/${config.repository}/git/commits`,
+    { body: { message, tree: treeSha, parents: [parentCommitSha] } }
+  );
+  const remoteCommitSha = commitRes.body.sha;
+
+  await githubApi.patchJson(
+    `/repos/${config.repository}/git/refs/heads/${branchName}`,
+    { body: { sha: remoteCommitSha, force: true } }
+  );
+
+  return remoteCommitSha;
+}
+
+async function pushFiles(
+  { branchName, files, message }: CommitFilesConfig,
+  { parentCommitSha, commitSha }: CommitResult
+): Promise<CommitSha | null> {
+  const additions: FileAddition[] = [];
+  const deletions: FileDeletion[] = [];
+
+  for (const file of files) {
+    if (file.type === 'addition') {
+      additions.push(file);
+    } else {
+      deletions.push(file);
+    }
+  }
+
+  const executableFiles = additions
+    .filter(({ isExecutable }) => isExecutable)
+    .map(({ path }) => path);
+  if (executableFiles.length) {
+    logger.warn(
+      { branchName, executableFiles },
+      'Platform-native commit: found executable files'
+    );
+  }
+
+  const fileChanges: GithubGraphqlFileChanges = {
+    additions: additions.map(({ path, contents }) => ({
+      path,
+      contents: Buffer.from(contents).toString('base64'),
+    })),
+    deletions: deletions.map(({ path }) => ({ path })),
+  };
+
+  const variables: GraphqlVariables = {
+    repo: config.repository,
+    repositoryId: config.repositoryId,
+    branchName: `refs/heads/${branchName}`,
+    oid: parentCommitSha,
+    fileChanges,
+    message,
+  };
+
+  try {
+    const { data, errors } = await githubApi.requestGraphql<{
+      createCommitOnBranch: { commit: { oid: string } };
+    }>(commitFilesMutation, { variables });
+
+    if (errors) {
+      if (errors.some(({ type }) => type === 'STALE_DATA')) {
+        return forcePushFiles(branchName, parentCommitSha, commitSha, message);
+      } else {
+        logger.warn(
+          { branchName, errors },
+          'Platform-native commit: GraphQL errors'
+        );
+        return null;
+      }
+    }
+
+    return data?.createCommitOnBranch?.commit?.oid;
+  } catch (err) {
+    logger.debug({ branchName, err }, 'Platform-native commit: unknown error');
+    return null;
+  }
+}
+
+export async function commitFiles(
+  config: CommitFilesConfig
+): Promise<CommitSha | null> {
+  const commitResult = await git.prepareCommit(config);
+  if (!commitResult) {
+    const { branchName, files } = config;
+    logger.debug(
+      { branchName, files: files.map(({ path }) => path) },
+      `Platform-native commit: unable to prepare for commit`
```

Comment:
Does this also happen because the commit doesn't change anything? In which case there'll be a debug message from git layer too?